src/ucs/datastruct/arbiter.c

```diff
@@ -94,7 +94,14 @@ void ucs_arbiter_group_push_head_elem_always(ucs_arbiter_t *arbiter,
 
     if (head->list.next != NULL) {
         ucs_assert(arbiter != NULL);
-        ucs_arbiter_group_head_replaced(arbiter, head, elem);
+        if (ucs_list_prev(&head->list, ucs_arbiter_elem_t, list) == head) {
+            arbiter->current = elem;
+            ucs_list_head_init(&elem->list);
+        } else {
+            ucs_arbiter_group_head_replaced(arbiter, head, elem);
```

Comment:
no need this code - ucs_arbiter_group_head_replaced is fixed now

Reply:
removed